File: src/ios/models/RelatedTracksByQueueRequest.swift
import Foundation

struct RelatedTracksByQueueRequest: Codable {
    let sources: [String]
    let albumTrackIds: [Int64]
    let excludeAlbumTrackIds: [Int64]
    let seedAlbumTrackIds: [Int64]

    private enum CodingKeys: String, CodingKey {
        case sources
        case albumTrackIds = "album_track_ids"
        case excludeAlbumTrackIds = "exclude_album_track_ids"
        case seedAlbumTrackIds = "seed_album_track_ids"
    }

    init(
        albumTrackIds: [Int64],
        excludeAlbumTrackIds: [Int64] = [],
        seedAlbumTrackIds: [Int64] = [],
        sources: [String] = ["cm", "stats", "playlists"]
    ) {
        self.sources = sources
        self.albumTrackIds = albumTrackIds
        self.excludeAlbumTrackIds = excludeAlbumTrackIds
        self.seedAlbumTrackIds = seedAlbumTrackIds
    }
}

File: src/android/com/kuackmedia/androidauto/models/RelatedTracksByQueueRequest.kt
package com.kuackmedia.androidauto.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RelatedTracksByQueueRequest(
    val sources: List<String> = listOf("cm", "stats", "playlists"),
    @Json(name = "album_track_ids") val albumTrackIds: List<Long>,
    @Json(name = "exclude_album_track_ids") val excludeAlbumTrackIds: List<Long> = emptyList(),
    @Json(name = "seed_album_track_ids") val seedAlbumTrackIds: List<Long> = emptyList()
)
